src/ts/subtitleutils.ts

```diff
@@ -54,6 +57,22 @@ export class SubtitleSwitchHandler {
     this.uimanager.getConfig().events.onUpdated.subscribe(this.refreshSubtitles);
   }
 
+  /**
+   * @param subtitleID (optional) If set, the according stored subtitle language will be set to active; If not set, the stored stored subtitle language will be set to inactive
+   */
+  public static setSubtitleLanguageStorage = (player: PlayerAPI, subtitleID?: string) => {
```

Comment:
This only gets called when any UI component is interacted with. In the case that the player would have subtitles already enabled when the UI is built or when something different from the UI enables a subtitle, this would not be affected.

I think the approach here would be to call this on component initialization with `player.subtitles.list().filter(sub => sub.enabled)` and then only reactively call this when the player fires a `SubtitleEnabled` or `SubtitleDisabled` event.

Reply:
Already works as intended.

Reply:
Today I learned :)